doc/smart-switch/smart-switch-database-architecture/smart-switch-database-design.md

```diff
@@ -0,0 +1,262 @@
+# Smart Switch Database design #
+
+## Table of Content 
+
+- [Smart Switch Database design](#smart-switch-database-design)
+  - [Table of Content](#table-of-content)
+    - [Revision](#revision)
+    - [Scope](#scope)
+    - [Definitions/Abbreviations](#definitionsabbreviations)
+    - [Overview](#overview)
+    - [Requirements](#requirements)
+    - [Architecture Design](#architecture-design)
+      - [Database services](#database-services)
+      - [Database flow](#database-flow)
+    - [High-Level Design](#high-level-design)
+    - [SAI API](#sai-api)
+    - [Configuration and management](#configuration-and-management)
+      - [CLI/YANG model Enhancements](#cliyang-model-enhancements)
+      - [Config DB Enhancements](#config-db-enhancements)
+    - [Warmboot and Fastboot Design Impact](#warmboot-and-fastboot-design-impact)
+    - [Memory Consumption](#memory-consumption)
+      - [DPU\_APPL\_DB](#dpu_appl_db)
+      - [DPU\_APPL\_STATE\_DB/DPU\_STATE\_DB](#dpu_appl_state_dbdpu_state_db)
+    - [Restrictions/Limitations](#restrictionslimitations)
+    - [Testing Requirements/Design](#testing-requirementsdesign)
+      - [Unit Test cases](#unit-test-cases)
+      - [System Test cases](#system-test-cases)
+    - [Open/Action items - if any](#openaction-items---if-any)
+
+### Revision
+
+|  Rev  | Date  | Author | Change Description               |
+| :---: | :---: | :----: | -------------------------------- |
+|  0.1  |       | Ze Gan | Initial version. Database design |
+
+### Scope
+
+This document provides a high-level design for Smart Switch database.
+
+### Definitions/Abbreviations 
+
+| Term | Meaning                           |
+| ---- | --------------------------------- |
+| NPU  | Network Processing Unit           |
+| DPU  | Data Processing Unit              |
+| DB   | Database                          |
+| GNMI | gRPC Network Management Interface |
+
+### Overview
+
+The Smart Switch comprises two integral components: the Network Processing Unit (NPU) and the Data Processing Unit (DPU), both operating on the SONiC OS. The database stack encompasses the entire database infrastructure for both the NPU and DPU. However, due to memory limitations on the DPU, certain overlayer objects, such as DASH objects, are stored in the NPU.
+
+In addition, dedicated database containers are maintained in the NPU for each DPU, serving the purpose of resource management within the Smart Switch architecture. This separation allows for efficient handling of database-related operations and ensures optimal utilization of resources across the entire Smart Switch.
+
+
+### Requirements
+
+- All databases, including those on both NPU and DPU, must be accessible through the GNMI server.
+- Each DPU database instance on the NPU is associated with a unique TCP port and domain Unix socket path.
+- All DPU database instances on the NPU will be bound to the IP address of the midplane bridge.
```

Comment:
>midplane bridge [](http://example.com/codeflow?start=79&length=15)

What is the definition of "midplane bridge"?

Reply:
Done. Add it in the Definitions/Abbreviations